src/components/Rating/ratingItems.js

```diff
@@ -3,15 +3,24 @@ import PropTypes from 'prop-types';
 import Star from './star';
 
 export default function RatingItems(props) {
-    const { onChange, value, name } = props;
+    const { onChange, value, name, readOnly } = props;
 
     return Array(5)
         .fill(0)
         .map((item, index) => {
             const key = `star-${index}`;
-            const filled = index < value;
+            const filled = index < Math.ceil(value);
+            const half = value < index + 1 && !Number.isInteger(Number(value));
```

Comment:
let's use isHalf and isFilled to express the intention of a boolean (normally we prefix all boolean variables names with is, are, or has)